### ISSUE TITLE
`@final` and `@override` diagnostics are not emitted if there are multiple reachable definitions

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ty#1677]
### Summary

Consider this snippet:

```py
from typing import final, override

def coinflip() -> bool:
    return False

class A:
    @final
    def method(self) -> None: ...

    if coinflip():
        @override
        def wut(self) -> None: ...
    else:
        @override
        def wut(self) -> None: ...

class B(A):
    if coinflip():
        def method(self) -> None: ...
    else:
        def method(self) -> None: ...
```

We should emit diagnostics on the definitions of `A.wut` and `B.method`, but we currently do not:
- `A.wut` is marked with `@override` but doesn't override anything
- `B.method` overrides a method decorated with `@final`

**Comment by AlexWaygood** [comment]
Note that we do emit these diagnostics if there is a single possibly-undefined definition:

```py
from typing import final, override

def coinflip() -> bool:
    return False

class A:
    @final
    def method(self) -> None: ...

    if coinflip():
        @override
        def wut(self) -> None: ...  # diagnostic correctly emitted here

class B(A):
    if coinflip():
        def method(self) -> None: ...  # diagnostic correctly emitted here
```
